build.psm1

```diff
@@ -0,0 +1,288 @@
+    }
+}
+
+
+function Uninstall-ScriptAnalyzer
+{
+    [CmdletBinding(SupportsShouldProcess)]
+    param ( $ModulePath = $(Join-Path -Path (Get-UserModulePath) -ChildPath PSScriptAnalyzer) )
+    END {
+        if ( $PSCmdlet.ShouldProcess("$modulePath") ) {
+            Remove-Item -Recurse -Path "$ModulePath" -Force
+        }
+    }
+}
+
+# install script analyzer, by default into the users module path
+function Install-ScriptAnalyzer
+{
+    [CmdletBinding(SupportsShouldProcess)]
+    param ( $ModulePath = $(Join-Path -Path (Get-UserModulePath) -ChildPath PSScriptAnalyzer) )
+    END {
+        if ( $PSCmdlet.ShouldProcess("$modulePath") ) {
+            Copy-Item -Recurse -Path "$destinationDir" -Destination "$ModulePath\." -Force
+        }
+    }
+}
+
+# if script analyzer is installed, remove it
+function Uninstall-ScriptAnalyzer
+{
+    [CmdletBinding(SupportsShouldProcess)]
+    param ( $ModulePath = $(Join-Path -Path (Get-UserModulePath) -ChildPath PSScriptAnalyzer) )
+    END {
+        if (Test-Path $ModulePath -and (Get-Item $ModulePath).PSIsContainer )
+        {
+            Remove-Item -Force -Recurse $ModulePath
+        }
+    }
+}
+
+# Clean up the build location
+function Remove-Build
+{
+    [CmdletBinding(SupportsShouldProcess=$true)]
+    param ()
+    END {
+        if ( $PSCmdlet.ShouldProcess("${destinationDir}")) {
+            if ( Test-Path ${destinationDir} ) {
+                Remove-Item -Force -Recurse ${destinationDir}
+            }
+        }
+    }
+}
+
+
+# Build documentation using platyPS
+function Build-Documentation
+{
+    $docsPath = Join-Path $projectRoot docs
+    $markdownDocsPath = Join-Path $docsPath markdown
+    $outputDocsPath = Join-Path $destinationDir en-US
+    $requiredVersionOfplatyPS = 0.9
+    $modInfo = new-object Microsoft.PowerShell.Commands.ModuleSpecification -ArgumentList @{ ModuleName = "platyps"; ModuleVersion = $requiredVersionOfplatyPS}
+    if ( $null -eq (Get-Module -ListAvailable -FullyQualifiedName $modInfo))
+    {
+        throw "Cannot find required minimum version $requiredVersionOfplatyPS of platyPS. Install via 'Install-Module platyPS'"
+    }
+    if (-not (Test-Path $markdownDocsPath))
+    {
+        throw "Cannot find markdown documentation folder."
+    }
+    Import-Module platyPS
+    if ( ! (Test-Path $outputDocsPath)) {
+        $null = New-Item -Type Directory -Path $outputDocsPath -Force
+    }
+    $null = New-ExternalHelp -Path $markdownDocsPath -OutputPath $outputDocsPath -Force
+}
+
+# build script analyzer (and optionally build everything with -All)
+function Build-ScriptAnalyzer
+{
+    [CmdletBinding(DefaultParameterSetName="BuildOne")]
+    param (
+        [Parameter(ParameterSetName="BuildAll")]
+        [switch]$All,
+
+        [Parameter(ParameterSetName="BuildOne")]
+        [ValidateSet("full", "core")]
+        [string]$Framework = "core",
+
+        [Parameter(ParameterSetName="BuildOne")]
+        [ValidateSet("3","4","5")]
```

Comment:
Should `"6"` be an option here? I feel like equating 5 with Core might prove confusing. We should probably have a comment to document it at least.

Reply:
Yes, and we should then remove the $Framework parameter and handle that internally since we know which ps version is core and which is full